src/integrations/HttpIntegration.ts

```diff
@@ -58,46 +58,70 @@ class HttpIntegration implements Integration {
         ThundraLogger.debug('<HTTPIntegration> Wrap');
 
         const nodeVersion = process.version;
-
         function wrapper(request: any) {
-            return function requestWrapper(options: any, callback: any) {
+            return function requestWrapper(a: any, b: any, c: any) {
                 let span: ThundraSpan;
+                const args = HTTPUtils.parseArgs(a, b, c);
+                const url = args.url;
+                let options = args.options;
+                const callback = args.callback;
                 try {
                     ThundraLogger.debug('<HTTPIntegration> Tracing HTTP request:', options);
 
                     const { tracer } = ExecutionContextManager.get();
-
                     if (!tracer) {
                         ThundraLogger.debug('<HTTPIntegration> Skipped tracing request as no tracer is available');
-                        return request.apply(this, [options, callback]);
+                        return request.apply(this, [a, b, c]);
+                    }
+
+                    let parsedUrl = url;
+                    if (typeof parsedUrl === 'string') {
+                        parsedUrl = Url.parse(parsedUrl);
                     }
 
-                    const method = (options.method || 'GET').toUpperCase();
-                    options = typeof options === 'string' ? url.parse(options) : options;
-                    const host = options.hostname || options.host || 'localhost';
-                    let path = options.path || options.pathname || '/';
+                    const host = (
+                        (parsedUrl && parsedUrl.hostname) ||
+                        (parsedUrl && parsedUrl.host) ||
+                        (options && options.hostname) ||
+                        (options && options.host) ||
+                        (options && options.uri && options.uri.hostname) ||
+                        'localhost'
+                    );
+
+                    if (callback && callback.__thundra_wrapped) {
```

Comment:
I think, you can move this logic to line `70` (just after `ThundraLogger.debug('<HTTPIntegration> Tracing HTTP request:', options);`);